### PR TITLE
Add auth checks to feed source summary

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Update to the feed source controller to apply two auth items:
- If a user can manage or view a feed
- If non-admin remove admin labels and notes.

Update to add the organization id and notes to the feed source summary.

The unit tests have been updated to as an admin.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]
Tested and looks good, thanks for the fix. We don't need the note IDs in the `feedSourceSummary` bc we hit the feedsource endpoint when a feedsource is selected, but  it's not the worst thing to keep them around for the future.

File: src/main/java/com/conveyal/datatools/manager/models/Label.java [hunk 1]
```diff
@@ -39,15 +39,24 @@ public String organizationId () {
...
     public Label (String name, String description, String color, boolean adminOnly, String projectId) {
+        this(null, name, description, color, adminOnly, projectId);
+    }
+
+    /**
+     * Create a new label with provided id.
+     */
+    public Label (String id, String name, String description, String color, boolean adminOnly, String projectId) {
         super();
+        if (id != null) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/Label.java]: Nit: I'd think the ctor with more params would call the ctor with fewer params.

File: src/main/java/com/conveyal/datatools/manager/models/Note.java [hunk 2]
```diff
@@ -32,6 +32,21 @@ public class Note extends Model implements Serializable {
     /** Whether the note should be visible to project admins only */
     public boolean adminOnly;
 
+    /**
+     * Create a new note with provided id.
+     */
+    public Note(String id, String body, boolean adminOnly) {
+        super();
+        if (id != null) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/Note.java]: Is there a need for a ctor that doesn't take an id?

File: src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java [hunk 3]
```diff
@@ -461,10 +509,33 @@ private static FeedVersion createFeedVersion(String id, String feedSourceId, Loc
...
+     * Helper method to create note.
+     */
+    private static Note createNote(String id, String body) {
+        Note note = new Note(id, body, false);
+        Persistence.notes.create(note);
+        return note;
+    }
+
+    /**
+     * Helper method to create label. If the id is provided save the label now if not deffer to test to save.
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java]: Typo "defer"

File: src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java [hunk 4]
```diff
@@ -366,53 +391,79 @@ public static FeedSource checkFeedSourcePermissions(Request req, FeedSource feed
...
+    }
+
+    /**
+     * Removes labels from a feed that a user is not allowed to view. Returns cleaned notes.
+     *
+     * @param labelIds The labels to clean.
+     * @param isAdmin  Is the user an admin? Changes what is returned.
+     * @return Labels the user is allowed to see.
+     */
+    protected static List<String> cleanFeedSourceLabelIdsForNonAdmins(List<String> labelIds, boolean isAdmin) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java:441]: Could `cleanFeedSourceLabelIdsForNonAdmins` and `cleanFeedSourceNotesForNonAdmins` be combined?

**Review by binh-dam-ibigroup — APPROVED** [review]
It works indeed!

File: src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java (hunk 4 above)
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java:441]: Although they are similar, the persistence type (labels/notes) is hard coded and can't be abstracted.

**Pull request merged**